### PR TITLE
feat: Adjust name of generated Python SDK

### PR DESCRIPTION
**Pull request by flaviuvadan**
This PR adds configurations to the OpenAPI generator so that the generated Python SDK uses `argo-workflows` as the name rather than `openapi-client`.

Issue: #7248 

Originally tested in: https://github.com/argoproj/argo-workflows/pull/7305

### CODE REVIEW

File: sdks/python/Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ WD                    := $(shell echo "`pwd`/client")
 DOCKER = docker run --rm -v $(WD):/wd --workdir /wd
 CHOWN = chown -R $(shell id -u):$(shell id -g)
 
-publish: generate
+publish:
```
Comment (terrytangyuan) [inline sdks/python/Makefile:15]: This was there for testing in the CI: https://github.com/argoproj/argo-workflows/blob/master/.github/workflows/sdks.yaml#L20

File: sdks/python/Makefile [hunk 2]
```diff
@@ -24,15 +24,21 @@ generate:
...
+		--model-name-suffix '' \
+		--artifact-id argo-python-client \
 		--global-property modelTests=false \
- 		--global-property packageName=workflows.client \
- 		--global-property packageVersion=${CLIENT_VERSION}
+ 		--global-property packageName=argo_workflows \
 		--generate-alias-as-model
 	# https://vsupalov.com/docker-shared-permissions/#set-the-docker-user-when-running-your-container
 	$(CHOWN) $(WD) || sudo $(CHOWN) $(WD)
+	rm -rf $(WD)/.gitlab-ci.yml $(WD)/.travis.yml $(WD)/git_push.sh
```
Comment (terrytangyuan) [inline sdks/python/Makefile:44]: Not needed since we have explicitly gitignored them?

**Review by terrytangyuan — COMMENTED** [review]
Can you make sure the generated SDK works as expected besides the known issue? https://github.com/argoproj/argo-workflows/issues/7293

**Comment by alexec** [comment]
You need to run `make pre-commit -B`

**Review by alexec — CHANGES_REQUESTED** [review]
Please run `make pre-commit -b` and commit the changes. We should then be good to go.

**Review by terrytangyuan — CHANGES_REQUESTED** [review]
Could you update the examples in README and the examples folder?

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**